core/observables/file.py

```diff
@@ -1,19 +1,37 @@
 from __future__ import unicode_literals
 
+from flask import url_for
+from flask_mongoengine.wtf import model_form
 from mongoengine import *
 
+
 from core.observables import Observable
-from core.observables import Hash
+from core.database import StringListField
 
 
 class File(Observable):
 
+    value = StringField(verbose_name="Value")
     mime_type = StringField(verbose_name="MIME type")
-    hashes = ListField(ReferenceField(Hash), verbose_name="Hashes")
-    body = FileField(verbose_name="File content")
+    hashes = DictField(verbose_name="Hashes")
+    body = ReferenceField("AttachedFile")
+    filenames = ListField(StringField(), verbose_name="Filenames")
```

Comment:
I am not seeing this. Not in the upload form, not in the detailed observable view, not in the API.